### PR TITLE
Add rectangle shape

### PR DESCRIPTION
**Pull request by mgautam98**
Added rectangle shape with test and docs

### CODE REVIEW

File: src/core.jl [hunk 1]
```diff
@@ -136,6 +136,21 @@ struct Cross <: Drawable
     range::UnitRange{Int}
 end
 
+
+"""
+    rectangle = Rectangle(p, x, y)
```
Comment (johnnychen94) [inline src/core.jl:141]: I prefer to not support `Rectangle(p::Point, x, y)` as the meaning of `x` and `y` can be confusing.

Instead, I prefer the following API:

```julia
Rectangle(from::Point, to::Point)
Rectangle(from::CartesianIndex, to::CaretesianIndex)
```

Comment (mgautam98) [reply src/core.jl:141]: Yes this seems more reasonable. I tried to follow [Matlab's API](https://www.mathworks.com/help/images/ref/images.roi.rectangle.html) but we can do better 😸

File: src/core.jl [hunk 2]
```diff
@@ -136,6 +136,21 @@ struct Cross <: Drawable
...
+
+"""
+    rectangle = Rectangle(p1, p2)
+A `Drawable` rectangle with p1 as top left and p2 as bottom right vertices.
+"""
+
+Rectangle(from::CartesianIndex{2}, to::CartesianIndex{2}) = 
+    Polygon([from,CartesianIndex(from[1],to[2]),to,CartesianIndex(to[1], from[2])])
+
+Rectangle(x1::Int, y1::Int, x2::Int, y2::Int) = Rectangle(CartesianIndex(x1, y1), CartesianIndex(x2, y2))
```
Comment (johnnychen94) [inline src/core.jl:148]: I suggest we remove this method as it's very confusing.

```julia
Rectangle(1, 2, 3, 4) # what does this mean?
```

File: test/paths.jl [hunk 3]
```diff
@@ -84,3 +84,14 @@ end
...
+
+
+@testset "Rectangle" begin
+    img = zeros(Gray, 10, 10)
+    expected = copy(img)
+    expected[2, 3:7] .= Gray(1)
+    expected[5, 3:7] .= Gray(1)
+    expected[2:5, 3] .= Gray(1)
+    expected[2:5, 7] .= Gray(1)
+    @test all(expected .== @inferred draw(img, Rectangle((2,3),(5,7)))) == true
```
Comment (johnnychen94) [inline test/paths.jl:96]: Also need to cover the cases for `CartesianIndex` and `Point` types.

**Comment by ashwani-rathee** [comment]
I can complete this,doesn't seem too difficult...should I give it a go?

**Comment by mgautam98** [comment]
> I can complete this,doesn't seem too difficult...should I give it a go?

Sure! give it a go

**Pull request closed without merging**